zerver/tornado/event_queue.py

```diff
@@ -1128,51 +1128,47 @@ def process_message_update_event(
 
     stream_name = event_template.get("stream_name")
     message_id = event_template["message_id"]
+    rendering_only_update = event_template["rendering_only"]
 
     for user_data in users:
         user_profile_id = user_data["id"]
 
-        if "user_id" in event_template:
-            # The user we'll get here will be the sender if the message's
-            # content was edited, and the editor for topic edits. That's
-            # the correct "acting_user" for both cases.
-            acting_user_id = event_template["user_id"]
-        else:
-            # Events without a `user_id` field come from the do_update_embedded_data
-            # code path, and represent just rendering previews; there should be no
-            # real content changes.
-            # It doesn't really matter what we set `acting_user_id` in this case,
-            # because we know this event isn't meant to send notifications.
-            acting_user_id = user_profile_id
```

Comment:
Since the `rendering_only` flag was added recently, we likely need to preserve some version of this old code with a `TODO/compatibility` comment, because a server upgrading directly from 4.0 to 6.0 would potentially have some events in the old format in its queue.

Reply:
(Usually, we try to the extent possible to contain that complexity in one place, e.g. having a block reformat the event to match the new format, so that most code can just handle the new format).

Reply:
I had an intuition that this was needed when I was working on this, so it's good to have that confirmed and know how to deal with that.